mantle/kola/tests/misc/network.go

```diff
@@ -293,35 +363,141 @@ var (
 `
 )
 
+func NetworkBondWithDhcp(c cluster.TestCluster) {
+	primaryMac := "52:55:00:d1:56:00"
+	secondaryMac := "52:55:00:d1:56:01"
+	primaryIp := "192.168.0.55"
+	secondaryIp := "192.168.0.56"
+
+	setupBondWithDhcpTest(c, primaryMac, secondaryMac, primaryIp, secondaryIp)
+
+	checkOvsBridge(c, primaryMac, primaryIp)
+}
+
+func initSetupVethPairsScript(primaryMac, secondaryMac, primaryIp, secondaryIp string) string {
+	return fmt.Sprintf(setupVethPairsTemplate, primaryMac, secondaryMac, primaryIp, secondaryIp)
+}
+
+func setupBondWithDhcpTest(c cluster.TestCluster, primaryMac, secondaryMac, primaryIp, secondaryIp string) {
+	var m platform.Machine
+	var err error
+	options := platform.QemuMachineOptions{}
+
+	setupVethPairs := initSetupVethPairsScript(primaryMac, secondaryMac, primaryIp, secondaryIp)
+
+	var userdata *conf.UserData = conf.Ignition(fmt.Sprintf(`{
+		"ignition": {
+			"version": "3.2.0"
+		},
+		"storage": {
+			"files": [
+				{
+					"path": "/etc/systemd/network/99-default.link",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 420
+				},
+				{
+					"path": "/etc/NetworkManager/conf.d/10-dhcp-config.conf",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 420
+				},
+				{
+					"path": "/usr/local/bin/capture-macs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				},
+				{
+					"path": "/usr/local/bin/create-veth-pairs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				},
+				{
+					"path": "/usr/local/bin/setup-ovs",
+					"contents": { "source": "data:text/plain;base64,%s" },
+					"mode": 755
+				}
+			]
+		},
+		"systemd": {
+			"units": [
+				{
+					"enabled": true,
+					"name": "openvswitch.service"
+				},
+				{
+					"contents": "[Unit]\nDescription=Capture MAC address from kargs\nAfter=network-online.target\nAfter=openvswitch.service\nConditionKernelCommandLine=macAddressList\n\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/capture-macs\n\n[Install]\nRequiredBy=multi-user.target\n",
+					"enabled": true,
+					"name": "capture-macs.service"
+				},
+				{
+					"contents": "[Unit]\nDescription=Create VETH Pairs and Configue OVS interface over bond\nAfter=network-online.target\nAfter=capture-macs.service\nAfter=openvswitch.service\n\n[Service]\nType=oneshot\nExecStart=/usr/local/bin/create-veth-pairs\n\n[Install]\nRequiredBy=multi-user.target\n",
+					"enabled": true,
+					"name": "create-veth-pairs.service"
+				}
+			]
+		}
+	}`,
+		base64.StdEncoding.EncodeToString([]byte(defaultLinkConfig)),
```

Comment:
Do we need all these scripts in this test?

Reply:
I was keeping them in order to be as close to the target configuration. If some of the scripts may interfere in the future in some unforeseeable way, we would catch it. 
But if you think it's not relevant in any way then I can remove.

Reply:
We should either test it knowing what is the overlap or not not at all. If some of those scripts take effect and influence the tested script, it's fine

Reply:
not sure I got you here, could you elaborate?

Reply:
Putting all the scripts on one pile to check that things work fine "just in case" sounds odd to me. I would like us to understand whether there is a possible overlap first and then make an educated decision. This set of tests does not replicate exactly what would happen in production, so it is almost as if we were testing integration between integration tests

Reply:
ack :)
From what I gather:
- `defaultLinkConfig` script is making sure that the nic are named according to their mac-address
since we are not relying on this in our tests, then I think we can drop it from both tests.
- `dhcpClientConfig` - we can drop it for multiple-nics test since we don't use dhcp there. in the dhcp test we should use it - only to make sure that we are consistently using the same dhcp client.
Done